src/helpers/BezierCurve.cpp

```diff
@@ -6,24 +6,21 @@
 #include <algorithm>
 
 void CBezierCurve::setup(std::vector<Vector2D>* pVec) {
-    m_dPoints.clear();
-
     const auto BEGIN = std::chrono::high_resolution_clock::now();
 
-    m_dPoints.emplace_back(Vector2D(0, 0));
-
-    for (auto const& p : *pVec) {
-        m_dPoints.push_back(p);
-    }
-
-    m_dPoints.emplace_back(Vector2D(1, 1));
+    // Avoid reallocations by reserving enough memory upfront
+    m_dPoints.resize(pVec->size() + 2);
+    m_dPoints[0] = Vector2D(0, 0); // Start point
+    std::copy(pVec->begin(), pVec->end(), m_dPoints.begin() + 1);
```

Comment:
dangerous, no bound checks. I'd rather just make a loop

Reply:
ok changed to a loop with bounds checking instead, pretty much same speedup still.